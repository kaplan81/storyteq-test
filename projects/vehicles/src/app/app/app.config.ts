import { provideHttpClient } from '@angular/common/http';
import {
  APP_INITIALIZER,
  ApplicationConfig,
  provideExperimentalZonelessChangeDetection,
} from '@angular/core';
import { provideRouter } from '@angular/router';
import { Config } from '@vehicles/app/models/config.model';
import { Observable } from 'rxjs';
import { routes } from './app.routes';
import { ConfigService } from './services/config/config.service';

export function appInitializerFactory(configService: ConfigService): () => Observable<Config> {
  return () => configService.loadConfig();
}

export const appConfig: ApplicationConfig = {
  providers: [
    provideExperimentalZonelessChangeDetection(),
    provideHttpClient(),
    {
      provide: APP_INITIALIZER,
      useFactory: appInitializerFactory,
      deps: [ConfigService],
      multi: true,
    },
    // provideRouter(routes, withDebugTracing()),
    provideRouter(routes),
  ],
};
