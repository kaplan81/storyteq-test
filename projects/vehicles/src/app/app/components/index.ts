export * from '@vehicles/app/components/spinner/spinner.component';
