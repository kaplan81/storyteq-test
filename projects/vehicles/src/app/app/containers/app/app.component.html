<!-- We do not apply BEM for app component -->
<header>
  <h1>{{ title }}</h1>
</header>
<main>
  <section>
    <router-outlet></router-outlet>
  </section>
</main>
<footer>
  <p>&copy; 2024 {{ title }}</p>
</footer>
