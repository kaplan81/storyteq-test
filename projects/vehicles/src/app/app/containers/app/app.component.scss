header {
  background-color: #3f577f;
  color: #fff;
  padding: 20px;
  text-align: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  z-index: 10;
}

main {
  min-height: calc(100vh - 130px);
  background-color: #f5f5f5;
  padding: 20px;
  padding-top: 80px;
  section {
    background-color: #fff;
    border-radius: 5px;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
    padding: 20px;
  }
}

footer {
  background-color: #3f577f;
  color: #fff;
  padding: 10px;
  text-align: center;
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  z-index: 10;
}
