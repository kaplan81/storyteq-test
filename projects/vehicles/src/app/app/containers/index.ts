export * from '@vehicles/app/containers/app/app.component';
