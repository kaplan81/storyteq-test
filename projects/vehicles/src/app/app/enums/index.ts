export * from '@vehicles/app/enums/breakpoint.enum';
export * from '@vehicles/app/enums/env.enum';
export * from '@vehicles/app/enums/routed.enum';
