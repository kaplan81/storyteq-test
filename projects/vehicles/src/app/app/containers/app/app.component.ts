import { Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { TitleMixin, emptyBase } from '@vehicles/app/mixins';
import { TitledComponent } from '@vehicles/app/models';

/**
 * Tell ngc about new properties.
 */
export interface AppComponent extends TitledComponent {}
@Component({
  selector: 'teq-root',
  standalone: true,
  imports: [RouterOutlet],
  templateUrl: 'app.component.html',
  styleUrl: 'app.component.scss',
})
export class AppComponent extends TitleMixin(emptyBase, 'Cars“R”Us') {}
