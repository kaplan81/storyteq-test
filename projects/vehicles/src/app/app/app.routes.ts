import { Routes } from '@angular/router';
import { CarsComponent } from '@vehicles/cars/containers';

export const routes: Routes = [{ path: '', component: CarsComponent }];
