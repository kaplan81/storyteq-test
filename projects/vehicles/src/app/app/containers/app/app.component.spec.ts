import { ComponentFixture, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
    }).compileComponents();
    fixture = TestBed.createComponent(AppComponent);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
