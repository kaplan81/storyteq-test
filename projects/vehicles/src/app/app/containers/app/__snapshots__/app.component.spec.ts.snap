// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<teq-root
  title={[Function String]}
>
  <header
    role="banner"
  >
    <h1>
      Cars“R”Us
    </h1>
  </header><main
    role="main"
  >
    <section>
      <router-outlet />
    </section>
  </main><footer
    role="contentinfo"
  >
    <p>
      © 2024 Cars“R”Us
    </p>
  </footer>
</teq-root>
`;
